packages/runtime/src/registry.tsx

```diff
@@ -1,20 +1,36 @@
 import { Alert } from "antd";
 import type { ReactElement } from "react";
 
+type WidgetComponent<T> = React.FC<T>;
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const mainRegistry: Record<string, WidgetComponent<any> | undefined> = {};
 // eslint-disable-next-line @typescript-eslint/no-explicit-any
-const registry: Record<string, React.FC<any> | undefined> = {};
+const backupRegistry: Record<string, WidgetComponent<any> | undefined> = {};
```

Comment:
What is the purpose of having two registries?

Reply:
When user created any custom widget with the same name of our in-build widgets, that time we can store our in-built widget into backupRegistry and load the custom widget into mainRegistry and serve the widget, and when screen gets destroyed, we remove that custom widget from mainRegistry and pick our in-built widget from the backupRegistry and load into mainRegistry